capsules/src/net/udp/driver.rs

```diff
@@ -466,154 +436,176 @@ impl<'a> LegacyDriver for UDPDriver<'a> {
     ///        before subscribe() is used to set up the recv callback. Additionally, apps can only
     ///        send on ports after they have bound to said port. If this command is called
     ///        and the address in rx_cfg is 0::0 : 0, this command will reset the option
-    ///        containing the bound port to None and set the rx callback to None. Notably,
+    ///        containing the bound port to None. Notably,
     ///        the current implementation of this only allows for each app to bind to a single
     ///        port at a time, as such an implementation conserves memory (and is similar
-    ///        to the approach applied by TinyOS and Riot). Further, there is
-    ///        currently no mechanism for anything in the kernel to bind to ports, and there
-    ///        is no distinction between ephemeral ports and reserved ports.
-    /// - `4`: Returns the maximum payload that can be transmitted by apps using this driver.
+    ///        to the approach applied by TinyOS and Riot).
+    ///        /// - `4`: Returns the maximum payload that can be transmitted by apps using this driver.
     ///        This represents the size of the payload buffer in the kernel. Apps can use this
     ///        syscall to ensure they do not attempt to send too-large messages.
 
-    fn command(&self, command_num: usize, arg1: usize, _: usize, appid: AppId) -> ReturnCode {
+    fn command(&self, command_num: usize, arg1: usize, _: usize, appid: AppId) -> CommandResult {
         match command_num {
-            0 => ReturnCode::SUCCESS,
+            0 => CommandResult::success(),
 
             //  Writes the requested number of network interface addresses
             // `arg1`: number of interfaces requested that will fit into the buffer
-            1 => self.do_with_cfg_mut(appid, arg1 * mem::size_of::<IPAddr>(), |cfg| {
-                let n_ifaces_to_copy = cmp::min(arg1, self.interface_list.len());
-                let iface_size = mem::size_of::<IPAddr>();
-                for i in 0..n_ifaces_to_copy {
-                    cfg[i * iface_size..(i + 1) * iface_size]
-                        .copy_from_slice(&self.interface_list[i].0);
-                }
-                // Returns total number of interfaces
-                ReturnCode::SuccessWithValue {
-                    value: self.interface_list.len(),
+            1 => {
+                let res = self
+                    .apps
+                    .enter(appid, |app, _| {
+                        app.app_cfg.mut_map_or(Err(ErrorCode::INVAL), |cfg| {
+                            if cfg.len() != arg1 * mem::size_of::<IPAddr>() {
```

Comment:
On a tangential note, it would be good to document what error codes are expected when. This could just as easily be an ESIZE.

Reply:
I added a couple more error code docs to what was already there -- completely fleshing them out requires looking at the possible return codes that can be created all the way down the networking stack, but I think that is outside the scope of this PR.